tycho-plugins/target-platform-utils/src/main/java/org/jboss/tools/tycho/targets/TargetToRepoMojo.java

```diff
@@ -129,77 +138,69 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 			final DestinationRepositoryDescriptor destinationDescriptor = new DestinationRepositoryDescriptor(this.outputRepository, this.sourceTargetFile.getName(), true, true,
 					false, false, true);
 
-	        List<IUDescription> initialIUs = new ArrayList<IUDescription>();
-	        for (final Location loc : target.getLocations()) {
-	        	if (loc instanceof InstallableUnitLocation) {
-	        		for (Unit unit : ((InstallableUnitLocation)loc).getUnits()) {
-	        			initialIUs.add(new IUDescription(unit.getId(), unit.getVersion()));
-	        		}
-	        	}
-	        }	        
-	        mirrorService.mirrorStandalone(sourceDescriptor, destinationDescriptor, initialIUs, createMirrorOptions(), new BuildOutputDirectory(this.project.getBuild().getOutputDirectory()));
+			TargetPlatformConfigurationStub tp = new TargetPlatformConfigurationStub();
+			tp.setEnvironments(ALL_SUPPORTED_TARGET_ENVIRONMENTS);
+			tp.addTargetDefinition(target);
+
+			this.p2Factory = this.equinox.getService(P2ResolverFactory.class);
+			P2Resolver tpResolver = this.p2Factory.createResolver(new MavenLoggerAdapter(this.logger, getLog().isDebugEnabled()));
+			tpResolver.setEnvironments(ALL_SUPPORTED_TARGET_ENVIRONMENTS);
+			for (final Location loc : target.getLocations()) {
+				if (loc instanceof InstallableUnitLocation) {
+					for (Unit unit : ((InstallableUnitLocation)loc).getUnits()) {
+						tpResolver.addDependency(ArtifactType.TYPE_INSTALLABLE_UNIT , unit.getId(), unit.getVersion());
+					}
+				}
+			}
+
+			P2ResolutionResult mirroredArtifacts = tpResolver.resolveMetadata(tp, this.executionEnvironment);
+			List<IUDescription> iusToMirror = new ArrayList<IUDescription>();
+			for (Entry entry : mirroredArtifacts.getArtifacts()) {
+				if (ArtifactType.TYPE_INSTALLABLE_UNIT.equals(entry.getType()) && !entry.getId().contains("a.jre.javase")) {
```

Comment:
`a.jre.javase` can be extracted in a constant

Reply:
I'd rather not see the constant definition in the code here. It should be somewhere else in Tycho, but I couldn't find a symbol for it. In such case, I prefer keeping the issue of "dereferencing a constant string" visible to more easily consider migrating to a constant definition when this is added in the appropriate class.